lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -228,6 +228,37 @@ class DecomposeAtenSelectIntOp : public OpRewritePattern<AtenSelectIntOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenNarrowOp : public OpRewritePattern<AtenNarrowOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenNarrowOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+    Value start = op.start();
+    Value dim = op.dim();
+    Value length = op.length();
+
+    Value one =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value startPlusLength =
+        rewriter.create<AtenAddIntOp>(loc, one.getType(), start, length);
+
+    // `aten.slice.tensor` doesn't squeeze the dim even when it's size 1 after
+    // slicing, while `aten.select.int` does.
+
+    // Replaces the AtenSliceTensorOp op, provides values from the nearly
+    // identical aten::narrow
```

Comment:
nit: This comment is not necessary. The code makes it very clear that we are replacing `AtenNarrowOp` with `AtenSliceTensorOp`